### PR TITLE
Scoped variables builder

### PR DESCRIPTION
**Pull request by yevbes**
#### Check List:
* Unit tests: YES / NO / NA
* Documentation: YES / NO / NA

### CODE REVIEW

File: modules/flowable-engine/src/main/java/org/flowable/engine/TaskService.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import java.util.Map;
 import java.util.Set;
 
+import org.flowable.task.api.TaskCompletionBuilder;
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/TaskService.java:22]: Please order the imports in alphabetic order as per project standards; thanks.

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/TaskCompletionBuilderImpl.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+package org.flowable.engine.impl;
+
+import java.util.Map;
+
+import org.flowable.common.engine.impl.interceptor.CommandExecutor;
+import org.flowable.engine.impl.cmd.CompleteTaskCmd;
+import org.flowable.engine.impl.cmd.CompleteTaskWithFormCmd;
+import org.flowable.task.api.TaskCompletionBuilder;
+import org.flowable.common.engine.api.variable.VariableCollectionsContainer;
+import org.flowable.common.engine.impl.variable.VariableCollectionsContainerImpl;
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/impl/TaskCompletionBuilderImpl.java:22]: Please order the imports in alphabetic order as per project standards; thanks.

File: modules/flowable-engine/src/test/java/org/flowable/engine/test/api/task/DelegateTaskTest.java [hunk 3]
```diff
@@ -68,7 +69,12 @@ public void testChangeCategoryInDelegateTask() {
             assertEquals("approval", task.getCategory());
             Map<String, Object> taskVariables = new HashMap<>();
             taskVariables.put("outcome", "approve");
-            taskService.complete(task.getId(), taskVariables, true);
+
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/test/java/org/flowable/engine/test/api/task/DelegateTaskTest.java:72]: FWIW, this is based on an older version of the file as line 68/69 is now:
<pre>
  assertThat(task.getCategory()).isEqualTo("approval");
</pre>
but it is still possible to merge your changes.  There is at least on other such file in this PR.

Comment (yevbes) [reply modules/flowable-engine/src/test/java/org/flowable/engine/test/api/task/DelegateTaskTest.java:72]: test class updated

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/TaskCompletionBuilderImpl.java (hunk 2 above)
Comment (yevbes) [reply modules/flowable-engine/src/main/java/org/flowable/engine/impl/TaskCompletionBuilderImpl.java]: done

File: modules/flowable-engine/src/main/java/org/flowable/engine/TaskService.java (hunk 1 above)
Comment (yevbes) [reply modules/flowable-engine/src/main/java/org/flowable/engine/TaskService.java:22]: done

**Comment by filiphr** [comment]
What is the difference between this PR and #2310?

**Comment by yevbes** [comment]
> What is the difference between this PR and #2310?

The difference is that in this PR I use Builder pattern approach to complete a task.

**Comment by jbarrez** [comment]
Manually merged this in https://github.com/flowable/flowable-engine/commit/c57f2c6198db1aaf47ab225015fe2351374f4fb3

**Pull request closed without merging**